### DISCUSSION TITLE
Webpack renaming class even with `minimize:false`

### DISCUSSION THREAD

**Discussion by trullock** [webpack/webpack#18415] [Q&A]
With this config

```
		optimization: {
			minimize: true,
			minimizer: [new TerserPlugin({
				terserOptions: {
					keep_classnames: true
				}
			})],
		},
```

one of my many classes is getting its name changed, from `MyClass` to `myClass_MyClass`, all other classes are untouched.

I thought this was a TerserPlugin bug but even with `minimize: false` the class is still getting renamed, but only in a production build.

I can't see any code that's especially different for `MyClass`... How can I debug this and/or why is it getting renamed so strangely?

This is my Babel loader code

```

				{
					test: /\.js$/,
					exclude: /node_modules/,
					use:
					{ 
						loader: "babel-loader",
						options: {
							// https://babeljs.io/docs/en/babel-preset-env
							// https://github.com/browserslist/browserslist
							presets: [
								["@babel/preset-env", { 
									"targets": //isDevelopment ?
										"last 1 chrome version" 
										//: "> 0.2%, not dead, not op_mini all" 
									}
								]
							],
							plugins: ["@babel/plugin-transform-class-properties", "@babel/plugin-transform-private-methods"]
						}
					}
				}
```

Thanks

**Comment by catester** [comment]
Even with latest version 5.93.0, I had the similar problem, when producing an ES module `minimize: false` is not respected (on other output types, it was respected).
Solution is to also set `minimizer: []`, setting to undefined/null does not work, only empty array [] works.

```
{
  mode: "production",
  optimization: {
    minimize: false,
    minimizer: []
  },
  output: {
    library: {
      type:  "module"
     },
  },
  experiments: {
    outputModule: true,
  }
}
```

**Reply by alexander-akait** [reply]
What do you mean? Please provide reprducible test repo, thank you

**Reply by catester** [reply]
I think this is the same error discussed in #15026

When I set `minimize: false` and use this setting to create a ES module bundle:

```
  output: {
    library: {
      type:  "module"
     },
  },
```

The generated bundle file is still minified (all line breaks are removed) so `minimize: false` is not respected. If also I set `minimizer: []`, then it's respected and the generated file is not minified.

However if I use another library type:

```
  output: {
    library: {
      type:  "global",
      name: "myLib"
     },
  },
```

Then `minimize: false`  alone is respected, and the generated file not minified.

Not a big deal, at least it can be fixed with the other property. The reason to use `minimize: false` for production files is that we have one file for "production unminified", and one file for "production minified".

**Reply by alexander-akait** [reply]
I don't see nothing related to https://github.com/webpack/webpack/discussions/15026, `minimize: false` and `output.library` completely unrelated options and do not affect each other in any way, please create reprducible test repo and I will look

**Reply by catester** [reply]
For reference, I used [webpack-stream](https://www.npmjs.com/package/webpack-stream) in gulp file to run Webpack. I don't know if it makes a difference, I examined webpack-stream code and did not notice anything related `minimize`, `minimizer`, `plugins`, it just passses the config to Webpack. When I run `npm ls  --depth=1` it shows I am indeed using the latest version `webpack@5.93.0`

```
└─┬ webpack-stream@7.0.0
  ├── fancy-log@1.3.3
  ├── lodash.clone@4.5.0
  ├── lodash.some@4.6.0
  ├── memory-fs@0.5.0
  ├── plugin-error@1.0.1 deduped
  ├── supports-color@8.1.1
  ├── through@2.3.8 deduped
  ├── vinyl@2.2.1
  └── webpack@5.93.0
```

Could it be when using `mode: "production"`, some plugins like TerserPlugin does not receive `minimize: false` setting? Ok I will try to create a repo when I have time.

**Reply by catester** [reply]
Ok I looked further into webpack-stream and found this code:

```
    // Cache compiler for future use
    const compiler = cache.compiler || webpack(config);
    cache.compiler = compiler;
```

This could explain the problem sorry for bugging you. As I run minify and unminified runs in same gulpfile, maybe it misbehaved because it caches webpack, looks suspicious

**Reply by alexander-akait** [reply]
Don't worry but we found the cause of the problem and it is good
